core/src/main/java/com/datastrato/gravitino/storage/relational/session/SqlSessions.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.storage.relational.session;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.TransactionIsolationLevel;
+
+/**
+ * SqlSessions is a utility class to maintain the MyBatis's {@link SqlSession} object. It is a
+ * thread local class and should be used to get the {@link SqlSession} object. It also provides the
+ * methods to commit, rollback and close the {@link SqlSession} object.
+ */
+public final class SqlSessions {
+  private static final ThreadLocal<SqlSession> sessions = new ThreadLocal<>();
+
+  private SqlSessions() {}
+
+  @VisibleForTesting
+  static ThreadLocal<SqlSession> getSessions() {
+    return sessions;
+  }
+
+  /**
+   * Get the SqlSession object. If the SqlSession object is not present in the thread local, then
+   * create a new SqlSession object and set it in the thread local.
+   *
+   * @return SqlSession object from the thread local storage.
+   */
+  public static SqlSession getSqlSession() {
+    SqlSession sqlSession = sessions.get();
+    if (sqlSession == null) {
+      sqlSession =
+          SqlSessionFactoryHelper.getInstance()
+              .getSqlSessionFactory()
+              .openSession(TransactionIsolationLevel.READ_COMMITTED);
+      sessions.set(sqlSession);
+      return sqlSession;
+    }
+    return sqlSession;
+  }
+
+  /**
+   * Commit the SqlSession object and close it. It also removes the SqlSession object from the
+   * thread local storage.
+   */
+  public static void commitAndCloseSqlSession() {
+    SqlSession sqlSession = sessions.get();
+    if (sqlSession != null) {
+      sqlSession.commit();
+      sqlSession.close();
+      sessions.remove();
+    }
+  }
+
+  /**
+   * Rollback the SqlSession object and close it. It also removes the SqlSession object from the
+   * thread local storage.
+   */
+  public static void rollbackAndCloseSqlSession() {
+    SqlSession sqlSession = sessions.get();
+    if (sqlSession != null) {
+      sqlSession.rollback();
+      sqlSession.close();
+      sessions.remove();
+    }
+  }
+
+  /** Close the SqlSession object and remove it from the thread local storage. */
+  public static void closeSqlSession() {
+    SqlSession sqlSession = sessions.get();
+    if (sqlSession != null) {
+      sqlSession.close();
+      sessions.remove();
```

Comment:
I can't get the point why do we remove the thread local session if we want to reuse the connection.

Reply:
First of all, the thread used to perform metadata operation requests on the server side is random. If we do not remove the corresponding `SqlSession` reference from ThreadLocal in time after completing the database operation, as long as the thread is not recycled, it will cause a memory leak. We should actively remove ThreadLocal variables after using them.
Secondly, database connections are maintained in a thread pool. `SqlSession` should not occupy a `Connection` for a long time. After completing the database operation, `close` and `remove` should be called so that `SqlSession` object can be recycled and the connection thread pool can recycle idle Connections. Otherwise, the connection pool and system resources may be exhausted.

Reply:
I can't get the point why we use thread local instead of a local variable if we clean thread local for every request.

Reply:
The reason why @xloya uses `ThreadLocal` is that he wants to hide the details of `SqlSession` and all `SqlSession` related operations lay in `SqlSessions`,  in the meanwhile, `ThreadLocal` can also guarantee thread safety. I have no preference on this point as long as it functions well.